ssl/statem/extensions.c

```diff
@@ -948,7 +948,6 @@ static int final_server_name(SSL *s, unsigned int context, int sent)
      * was successful.
      */
     if (s->server) {
-        /* TODO(OpenSSL1.2) revisit !sent case */
         if (sent && ret == SSL_TLSEXT_ERR_OK && !s->hit) {
             /* Only store the hostname in the session if we accepted it. */
```

Comment:
Yes

Reply:
It sounds like you're okay with that.
I can tolerate it, but given that ```SSL_get_servername()``` will never give it back, having ```SSL_SESSION_get0_hostname()``` return something different than ```SSL_get_servername()``` seems weird.  Doubly so that we let you get the SNI from the session on initial handshakes but not resumptions, even when for TLS 1.3 it's always a property of the connection, regardless of whether it's an initial or resumption connection.